### PR TITLE
Markdown footnote support

### PR DESCRIPTION
**Pull request by nswaldman**
This PR is intended to add footnote support in markdown fields (crucially, for the notes fields).

### CODE REVIEW

File: demoscene/templatetags/safe_markdown.py [hunk 1]
```diff
@@ -10,25 +10,28 @@
...
-md = markdown.Markdown(extensions=['nl2br', autolink])
+md = markdown.Markdown(extensions=['footnotes', 'nl2br', autolink])
 
 
 @register.filter(is_safe=True)
 def safe_markdown(value, arg=''):
     return mark_safe(
         cleaner.clean(
-            md.reset().convert(value)
+        md.reset().convert(value)
```
Comment (nswaldman) [inline demoscene/templatetags/safe_markdown.py:35]: fix indentation

File: demoscene/templatetags/safe_markdown.py [hunk 2]
```diff
@@ -10,25 +10,28 @@
 
 cleaner = Cleaner(
```
Comment (nswaldman) [inline demoscene/templatetags/safe_markdown.py:11]: Additions to the whitelist here are necessary for markdown footnotes to survive Bleach's Cleaner. Without it:

1. The `div.footnotes` container would be stripped (no easy styling crutch)
2. Links and backlinks couldn't work (no fragment ids on `li` and `sup`)

Comment (gasman) [reply demoscene/templatetags/safe_markdown.py:11]: Good call to bring this up! I'm fine with allowing all `div`s with any class attribute. Ideally markdown would have a built-in sanitisation step that would kick in for arbitrary tags in the source text, while still letting through tags that are generated by our own trusted code. But I guess no-one has seen the need to implement that yet :-(

I think the most evil thing someone can do with `<div class="...">` is replicate UI elements from elsewhere in our CSS, and I think we can live with that risk.

**Review by gasman — APPROVED** [review]
Looks good to me! Tried it, and all works nicely.

**Comment by gasman** [comment]
Merged in 9169d8ee7fc9ecc51e1a43f248f1f31f6dc0d4cc

**Comment by gasman** [comment]
Putting this here for when we inevitably go "wasn't there some discussion about replacing Bleach a few months ago"... https://daniel.feldroy.com/posts/2023-06-converting-from-bleach-to-nh3

Looks like nh3 [gives us a few smarter options for handling attributes](https://nh3.readthedocs.io/en/latest/) that would enable letting `<div class="footnotes">` through and not others.

**Pull request closed without merging**